### PR TITLE
Rename colorB -> color; colorA -> labelColor; test on [endpoint dynamic static]

### PR DESCRIPTION
**Pull request by mbarkhau**
This makes it possible to override the parameters `color` and `labelColor`.

ref #2673

### CODE REVIEW

File: core/base-service/coalesce-badge.js [hunk 1]
```diff
@@ -41,19 +41,34 @@ module.exports = function coalesceBadge(
   defaultBadgeData,
   { category, _cacheLength: defaultCacheSeconds } = {}
 ) {
+  // The "overrideX" naming is based on the endpoint badge, where these are
+  // indeed overrides for values supplied by an endpoint. In the case of the
```
Comment (paulmelnikow) [inline core/base-service/coalesce-badge.js]: In general, "override" here refers to values specified in the query string taking precedence over values provided by the service.

File: core/base-service/coalesce-badge.js [hunk 2]
```diff
@@ -41,19 +41,34 @@ module.exports = function coalesceBadge(
...
+    labelColor: overrideLabelColor,
   } = overrides
+
+  // Only use the legacy properties of the new ones are not provided
+  if (typeof overrideColor === 'undefined') {
+    overrideColor = legacyOverrideColor
+  }
+  if (typeof overrideLabelColor === 'undefined') {
+    overrideLabelColor = legacyOverrideLabelColor
+  }
```
Comment (paulmelnikow) [inline core/base-service/coalesce-badge.js]: Could you add unit tests in `coalesce-badge.spec.js`?

File: frontend/pages/endpoint.js [hunk 3]
```diff
@@ -173,11 +173,12 @@ const EndpointPage = () => (
...
         Default: <code>lightgrey</code>. The right color. Supports the eight
         named colors above, as well as hex, rgb, rgba, hsl, hsla and css named
-        colors.
+        colors. This can be overridden by the query string.
       </dd>
       <dt>labelColor</dt>
       <dd>
-        Default: <code>grey</code>. The left color.
+        Default: <code>grey</code>. The left color. This can be overridden by
+        the query string.
```
Comment (paulmelnikow) [inline frontend/pages/endpoint.js]: 👍

File: services/endpoint/endpoint.tester.js [hunk 4]
```diff
@@ -170,6 +170,20 @@ t.create('Invalid schema (mocked)')
...
+      .reply(200, {
+        schemaVersion: 1,
+        label: '',
+        message: 'yo',
+        color: 'blue',
+      })
+  )
+  .expectJSON({ name: '', value: 'yo', color: '#101010' })
+
+t.create('User legacy color overrides success color')
```
Comment (paulmelnikow) [inline services/endpoint/endpoint.tester.js]: Thanks for adding these tests!

**Comment by paulmelnikow** [comment]
I've updated the title here to hopefully catch the attention of folks who might be interested in this.

File: core/base-service/coalesce-badge.js (hunk 2 above)
Comment (mbarkhau) [reply core/base-service/coalesce-badge.js]: Added.

File: core/base-service/coalesce-badge.js (hunk 1 above)
Comment (mbarkhau) [reply core/base-service/coalesce-badge.js]: Reworded.

**Comment by mbarkhau** [comment]
Thank you for the feedback. I've updated the PR and hope I didn't miss anything.

File: core/base-service/coalesce-badge.spec.js [hunk 5]
```diff
@@ -92,11 +104,25 @@ describe('coalesceBadge', function() {
...
     it('converts a query-string numeric color to a string', function() {
+      expect(
+        coalesceBadge(
+          // Scoutcamp converts numeric query params to numbers.
+          { color: 123 },
+          { color: 'green' },
+          {}
+        ).color
+      ).to.equal('123')
+      // also expected for legacy name
```
Comment (paulmelnikow) [inline core/base-service/coalesce-badge.spec.js]: 👍

File: core/base-service/coalesce-badge.js (hunk 2 above)
Comment (paulmelnikow) [reply core/base-service/coalesce-badge.js]: Thanks!

File: frontend/components/usage.js [hunk 6]
```diff
@@ -271,7 +271,7 @@ export default class Usage extends React.PureComponent {
             >
               $.DATA.SUBDATA
             </a>
-            &gt;&amp;colorB=&lt;COLOR&gt;&amp;prefix=&lt;PREFIX&gt;&amp;suffix=&lt;SUFFIX&gt;
+            &gt;&amp;color=&lt;COLOR&gt;&amp;prefix=&lt;PREFIX&gt;&amp;suffix=&lt;SUFFIX&gt;
```
Comment (paulmelnikow) [inline frontend/components/usage.js]: 👍

**Comment by paulmelnikow** [comment]
@badges/shields-core Any concerns with this?

**Comment by PyvesB** [comment]
Nope, none, the change makes sense to me!

**Comment by paulmelnikow** [comment]
Thanks so much for your work on this @mbarkhau!

**Comment by calebcartwright** [comment]
Late to the party here but no concerns either! @mbarkhau want to echo the thanks and appreciation, well done! 

✨ 👏 💯

**Comment by mbarkhau** [comment]
Thanks for all the support :100:

**Pull request merged**